packages/devtools/lib/src/main.dart

```diff
@@ -132,71 +130,94 @@ class PerfToolFramework extends Framework {
     }
     return true;
   }
 
   Future<bool> isProfileBuild() async {
     try {
       final Isolate isolate = await serviceManager.service
-          .getIsolate(serviceManager.isolateManager.selectedIsolate.id);
+          .getIsolate(serviceManager.isolateManager.isolates.first.id);
+      // This evaluate statement will throw an error in a profile build.
       await serviceManager.service.evaluate(
-        serviceManager.isolateManager.selectedIsolate.id,
+        serviceManager.isolateManager.isolates.first.id,
         isolate.rootLib.id,
         '1+1',
       );
       // If we reach this return statement, no error was thrown and this is not
       // a profile build.
       return false;
-    } catch (e) {
-      if (e is RPCError) {
-        return true;
-      }
+    } on RPCError catch (_) {
+      return true;
     }
-    return false;
   }
 
   Future<void> addScreens() async {
     final _isFlutterApp = await isFlutterApp();
+    final _isFlutterWebApp = await isFlutterWebApp();
     final _isProfileBuild = await isProfileBuild();
 
     addScreen(InspectorScreen(
-      disabled: !_isFlutterApp || _isProfileBuild,
-      disabledTooltip: !_isFlutterApp
-          ? 'This section is disabled because you are not running a Flutter '
+      disabled: !_isFlutterApp || !_isFlutterWebApp || _isProfileBuild,
+      disabledTooltip: (!_isFlutterApp || !_isFlutterWebApp)
+          ? 'This screen is disabled because you are not running a Flutter '
               'application'
-          : 'This section is disabled because you are running a profile build '
+          : 'This screen is disabled because you are running a profile build '
           'of your application',
     ));
     addScreen(TimelineScreen(
-      disabled: !_isFlutterApp,
-      disabledTooltip: 'This section is disabled because you are not running a '
+      disabled: !_isFlutterApp || _isFlutterWebApp,
+      disabledTooltip: _isFlutterWebApp
+          ? 'This screen is disabled because it is not yet ready for Flutter'
+              ' Web'
+          : 'This screen is disabled because you are not running a '
           'Flutter application',
     ));
-    addScreen(MemoryScreen());
+    addScreen(MemoryScreen(
+      disabled: _isFlutterWebApp,
+      disabledTooltip:
+          'This screen is disabled because it is not yet ready for Flutter'
+          ' Web',
+    ));
     if (showPerformancePage) {
       addScreen(PerformanceScreen());
     }
+
     addScreen(DebuggerScreen(
-      disabled: _isProfileBuild || tabDisabledByQuery('debugger'),
-      disabledTooltip: _isProfileBuild
-          ? 'This section is disabled because you are running a profile build '
-              'of your application'
-          : 'This section is disabled because it provides functionality already'
-          ' available in your code editor',
+      disabled:
+          _isFlutterWebApp || _isProfileBuild || tabDisabledByQuery('debugger'),
+      disabledTooltip: _isFlutterWebApp
```

Comment:
add a helper method and make this more straightline logic. These conditionals and the indentation make this harder to read than it should be.

E.g. the helper should look like
if (_isFlutterWebApp) return 'foo';
if (_isProfileBuild) return 'bar';
return 'default;

Reply:
done.